src/App.js

```diff
@@ -96,6 +98,39 @@ type RouteInformation = {
   equipmentInfoId: ?string,
 };
 
+function getRouteInformation(props: Props): ?RouteInformation {
+  const location = props.location;
+  const allowedResourceNames = ['nodes', 'categories', 'search'];
+  const match = location.pathname.match(/(?:\/beta)?\/?(?:(-?\w+)(?:\/([-\w\d]+)(?:\/([-\w\d]+)(?:\/([-\w\d]+))?)?)?)?/i);
```

Comment:
Not a fan of this, but I see no better solution immediately, so lets do this :)

Reply:
This grew a bit too long ;)